Sources/SwiftFrameCore/Config/DeviceData.swift

```diff
@@ -0,0 +1,122 @@
+import AppKit
+import Foundation
+
+public struct DeviceData: Decodable, ConfigValidatable {
+
+    // MARK: - Properties
+
+    private let kScreenshotExtensions = Set<String>(arrayLiteral: "png", "jpg", "jpeg")
+
+    let outputSuffix: String
+    let templateImagePath: LocalURL
+    private let screenshotsPath: LocalURL
+
+    internal private(set) var screenshotsGroupedByLocale: [String: [String: URL]]!
```

Comment:
One thing I'm wondering is if it would be worth to merge `process` into `init` (here and elsewhere). Those `private(set)` force-unwrapped or empty-initialized properties ring my alarm bells a bit because using them before `process` is actually called would be a failure but the code doesn't really prevent that.

Reply:
Wasn't that the reason why we moved to the `process` step in the first place? If we do all that stuff in the `init` we have to provide a custom `init(from decoder: Decoder) throws` for each type

Reply:
Hm, that is a good point. We could potentially circumvent that by using lazy properties, subclasses or a composed type (e.g. one struct that parses the bare JSON data which is then fed into (or decorated) with another struct that processes the bare data and builds those computed values.

It's not entirely bad to have separate `init` and `process` methods on each type. It just felt a little suboptimal having to have force-unwrapped or empty-initialized properties that can only be used after both methods have been called.